client/src/components/ViewContact.jsx

```diff
@@ -86,36 +66,30 @@ const ViewContact = ({ selectedContactId, setContacts }) => {
         }
       );
 
+      if (!response.ok) {
+        throw new Error("Failed to edit contact");
+      }
+
       await response.json();
 
       const refreshedListData = await fetchContacts();
       setContacts(refreshedListData);
-
-      if (!response.ok) {
-        throw new Error("Failed to edit contact");
-      }
     } catch (error) {
       console.error("Error updating entry: ", error);
+
       setSnackbarMessage("Error editing contact. Please try again.");
-      setContactData((previousData) => ({
-        ...previousData,
-        newName: previousData.originalData.name,
-        newEmail: previousData.originalData.email,
-        newPhone: previousData.originalData.phone,
-        newNotes: previousData.originalData.notes,
-      }));
```

Comment:
I deleted this state setting code because I feel like it would be bad UX. I feel like most people would prefer the system kept their edits so they wouldn't have to re-edit their form data before trying to submit it again.